core/src/main/java/org/apache/druid/data/input/impl/JsonReader.java

```diff
@@ -90,11 +91,12 @@
   @Override
   protected List<InputRow> parseInputRows(String intermediateRow) throws IOException, ParseException
   {
+    List<InputRow> inputRows;
     try (JsonParser parser = new JsonFactory().createParser(intermediateRow)) {
       final MappingIterator<JsonNode> delegate = mapper.readValues(parser, JsonNode.class);
-      return FluentIterable.from(() -> delegate)
-                           .transform(jsonNode -> MapInputRowParser.parse(inputRowSchema, flattener.flatten(jsonNode)))
-                           .toList();
+      inputRows = FluentIterable.from(() -> delegate)
+                                               .transform(jsonNode -> MapInputRowParser.parse(inputRowSchema, flattener.flatten(jsonNode)))
+                                               .toList();
```

Comment:
nit: this formatting seems off (intellij moves it in line with the `.from` when i pull your branch locally)

Reply:
Fixed.